contracts/0.4.24/lib/StakeLimitUtils.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: 2022 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.4.24;
+
+//
+// We need to pack four variables into the same 256bit-wide storage slot
+// to lower the costs per each staking request.
+//
+// As a result, slot's memory aligned as follows:
```

Comment:
Need to mention that there's a "staking paused" state encoded by all fields being zero, and "staking unlimited" state encoded by `maxStakeLimit` being zero and `prevStakeBlockNumber` being non-zero.